examples/ensemble/plot_adaboost_multiclass.py

```diff
@@ -1,122 +1,242 @@
 
-n_trees_discrete = len(bdt_discrete)
-n_trees_real = len(bdt_real)
 
-# Boosting might terminate early, but the following arrays are always
-# n_estimators long. We crop them to the actual number of trees here:
-discrete_estimator_errors = bdt_discrete.estimator_errors_[:n_trees_discrete]
-real_estimator_errors = bdt_real.estimator_errors_[:n_trees_real]
-discrete_estimator_weights = bdt_discrete.estimator_weights_[:n_trees_discrete]
+weak_learners_misclassification_error = misclassification_error(
+    y_test, weak_learner.fit(X_train, y_train).predict(X_test)
+)
 
-plt.figure(figsize=(15, 5))
+dummy_classifiers_misclassification_error = misclassification_error(
+    y_test, dummy_clf.fit(X_train, y_train).predict(X_test)
+)
 
-plt.subplot(131)
-plt.plot(range(1, n_trees_discrete + 1), discrete_test_errors, c="black", label="SAMME")
-plt.plot(
-    range(1, n_trees_real + 1),
-    real_test_errors,
-    c="black",
-    linestyle="dashed",
-    label="SAMME.R",
+print(
+    "DecisionTreeClassifier's misclassification_error:"
+    f"{weak_learners_misclassification_error}\n"
+    "DummyClassifier's misclassification_error:"
+    f"{round(dummy_classifiers_misclassification_error, 3)}"
 )
-plt.legend()
-plt.ylim(0.18, 0.62)
-plt.ylabel("Test Error")
-plt.xlabel("Number of Trees")
 
-plt.subplot(132)
+# %%
+# After training the :class:`~sklearn.tree.DecisionTreeClassifier` model, the
+# achieved error surpasses the expected value that would have been obtained by
+# guessing the most frequent class label, as the
+# :class:`~sklearn.dummy.DummyClassifier` does.
+#
+# Now, we calculate the `misclassification_error` of the additive model
+# (:class:`~sklearn.tree.DecisionTreeClassifier`) at each boosting iteration to
+# assess its performance. To obtain the predictions at each iteration, we use
+# the :func:`~sklearn.ensemble._weight_boosting.staged_predict` method which
+# returns a generator containing the predictions of all the weak learners during
+# the training progresses. We compare these predictions to the actual values in
+# `y_test` by calculating the difference. This process allows us to generate the
+# data needed for plotting a learning curve.
+import matplotlib.pyplot as plt
+import pandas as pd
+
+boosting_errors = pd.DataFrame(
+    {
+        "Number of trees": range(1, n_estimators + 1),
+        "AdaBoost": [
+            misclassification_error(y_test, y_pred)
+            for y_pred in adaboost_clf.staged_predict(X_test)
+        ],
+    }
+).set_index("Number of trees")
+ax = boosting_errors.plot()
+ax.set_ylabel("Misclassification error on test set")
+ax.set_title("Convergence of AdaBoost algorithm")
+
 plt.plot(
-    range(1, n_trees_discrete + 1),
-    discrete_estimator_errors,
-    "b",
-    label="SAMME",
-    alpha=0.5,
+    [boosting_errors.index.min(), boosting_errors.index.max()],
+    [weak_learners_misclassification_error, weak_learners_misclassification_error],
+    color="c",
```

Comment:
I would prefer to have a high contrast between those the horizontal line and the adaboost error

```suggestion
    color="tab:orange",
```